### PR TITLE
Report function name in stack trace annotations

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117459

When working with internal flows, it can sometimes be ambiguous what
version of the code they are working with.  In this case, having the
function name available in the stack trace can help identify what you
are looking at.

Example now looks like:

```
[DEBUG]         # File: /data/users/ezyang/a/pytorch/a.py:5 in f, code: return x + x
```

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: torch/fx/graph.py [hunk 1]
```diff
@@ -284,11 +283,20 @@ class _PyTreeInfo(NamedTuple):
     in_spec: pytree.TreeSpec
     out_spec: Optional[pytree.TreeSpec]
 
+@dataclass(frozen=True)
+class _ParsedStackTrace:
+    """
+    Represents the top-most frame of a parsed stack trace
+    """
+    file: str
+    lineno: str
```
Comment (Skylion007) [inline torch/fx/graph.py:292]: Not suer how the lineno is parsed, but should it be typed as an int?

**Review by Skylion007 — APPROVED** [review]
Love PRs that improve the typing of a `namedtuple`

File: torch/fx/graph.py (hunk 1 above)
Comment (ezyang) [reply torch/fx/graph.py:292]: It's just interpolated into another string, s'fine ig

**Pull request closed without merging**